### DISCUSSION TITLE
openshift-install-linux-4.13.0-0.okd-2023-07-09-062029 fails installation

### DISCUSSION THREAD

**Discussion by jgforbes** [okd-project/okd#1655] [General]
**Describe the bug**
openshift-install for openshift 4.13.4 works perfectly
Latest okd installer fails with same configuration file.

**Version**
openshift-install-linux-4.13.0-0.okd-2023-07-09-062029

**How reproducible**
100%

**Log bundle**
]$ ./openshift-install create cluster --dir ~/okdInstallation
INFO Consuming Install Config from target directory 
INFO Obtaining RHCOS image file from 'https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/37.20221127.3.0/x86_64/fedora-coreos-37.20221127.3.0-vmware.x86_64.ova?sha256=' 
INFO The file was found in cache: /home/okd/.cache/openshift-installer/image_cache/fedora-coreos-37.20221127.3.0-vmware.x86_64.ova. Reusing... 
INFO Creating infrastructure resources...         
INFO Waiting up to 20m0s (until 9:48AM) for the Kubernetes API at https://api.amds-kub.jhuapl.edu:6443... 
INFO API v1.26.4-2868+a7ee68b55354d8-dirty up     
INFO Waiting up to 30m0s (until 10:03AM) for bootstrapping to complete... 
WARNING Failed to extract host addresses: could not extract IP with bootstrap MOID:  
INFO Skipping VM console logs gather: no gather methods registered for "vsphere" 
INFO Pulling debug logs from the bootstrap machine 

WARNING Unable to stat /openshift-install-linux-4.13.0-0.okd-2023-07-09-062029home/okd/okdInstallation/serial-log-bundle-20230710100339.tar.gz, skipping 
ERROR Cluster operator authentication Degraded is True with IngressStateEndpoints_MissingSubsets::OAuthClientsController_SyncError::OAuthServerDeployment_PreconditionNotFulfilled::OAuthServerRouteEndpointAccessibleController_SyncError::OAuthServerServiceEndpointAccessibleController_SyncError::OAuthServerServiceEndpointsEndpointAccessibleController_SyncError::WellKnownReadyController_SyncError: IngressStateEndpointsDegraded: No subsets found for the endpoints of oauth-server 
ERROR OAuthClientsControllerDegraded: no ingress for host oauth-openshift.apps.amds-kub.jhuapl.edu in route oauth-openshift in namespace openshift-authentication 
ERROR OAuthServerDeploymentDegraded: waiting for the oauth-openshift route to contain an admitted ingress: no admitted ingress for route oauth-openshift in namespace openshift-authentication 
ERROR OAuthServerDeploopenshift-install-linux-4.13.0-0.okd-2023-07-09-062029ymentDegraded:               
ERROR OAuthServerRouteEndpointAccessibleControllerDegraded: route "openshift-authentication/oauth-openshift": status does not have a valid host address 
ERROR OAuthServerServiceEndpointAccessibleControllerDegraded: Get "https://172.30.26.201:443/healthz": dial tcp 172.30.26.201:443: connect: connection refused 
ERROR OAuthServerServiceEndpointsEndpointAccessibleControllerDegraded: oauth service endpoints are not ready 
ERROR WellKnownReadyControllerDegraded: failed to get oauth metadata from openshift-config-managed/oauth-openshift ConfigMap: configmap "oauth-openshift" not found (check authentication operator, it is supposed to create this) 
ERROR Cluster operator authentication Available is False with OAuthServerDeployment_PreconditionNotFulfilled::OAuthServerServiceEndpointAccessibleController_EndpointUnavailable::OAuthServerServiceEndpointsEndpointAccessibleController_ResourceNotFound::ReadyIngressNodes_NoReadyIngressNodes::WellKnown_NotReady: OAuthServerServiceEndpointAccessibleControllerAvailable: Get
 "https://172.30.26.201:443/healthz": dial tcp 172.30.26.201:443: connect: connection refused 
ERROR OAuthServerServiceEndpointsEndpointAccessibleControllerAvailable: endpoints "oauth-openshift" not found 
ERROR ReadyIngressNodesAvailable: Authentication requires functional ingress which requires at least one schedulable and ready node. Got 0 worker nodes, 3 master nodes, 0 custom target nodes (none are schedulable or ready for ingress pods). 
ERROR WellKnownAvailable: The well-known endpoint is not yet available: failed to get oauth metadata from openshift-config-managed/oauth-openshift ConfigMap: configmap "oauth-openshift" not found (check authentication operator, it is supposed to create this) 
INFO Cluster operator baremetal Disabled is False with :  
INFO Cluster operator cloud-controller-manager TrustedCABundleControllerControllerAvailable is True with AsExpected: Trusted CA Bundle Controller works as expected 
INFO Cluster operator cloud-controller-manager TrustedCABundleControllerControllerDegraded is False with AsExpected: Trusted CA Bundle Controller works as expected 
INFO Cluster operator cloud-controller-manager CloudConfigControllerAvailable is True with AsExpected: Cloud Config Controller works as expected 
INFO Cluster operator cloud-controller-manager CloudConfigControllerDegraded is False with AsExpected: Cloud Config Controller works as expected 
INFO Cluster operator cloud-controller-manager CloudControllerOwner is True with AsExpected: Cluster Cloud Controller Manager Operator owns cloud controllers at 4.13.0-0.okd-2023-07-09-062029 
ERROR Cluster operator cluster-autoscaler Degraded is True with MissingDependency: machine-api not ready 
ERROR Cluster operator console Available is False with RouteHealth_RouteNotAdmitted: RouteHealthAvailable: console route is not admitted 
INFO Cluster operator etcd RecentBackup is Unknown with ControllerStarted: The etcd backup controller is starting, and will decide if recent backups are availablopenshift-install-linux-4.13.0-0.okd-2023-07-09-062029e or if a backup is required 
ERROR Cluster operator ingress Available is False with IngressUnavailable: The "default" ingress controller reports Available=False: IngressControllerUnavailable: One or more status conditions indicate unavailable: DeploymentAvailable=False (DeploymentUnavailable: The deployment has Available status condition set to False (reason: MinimumReplicasUnavailable) with message: Deployment does not have minimum availability.) 
INFO Cluster operator ingress Progressing is True with Reconciling: ingresscontroller "default" is progressing: IngressControllerProgressing: One or more status conditions indicate progressing: DeploymentRollingOut=True (DeploymentRollingOut: Waiting for router deployment rollout to finish: 0 of 2 updated replica(s) are available... 
INFO ).                                           
INFO Not all ingress controllers are available.   
ERROR Cluster operator ingress Degraded is True with IngressDegraded: The "default" ingress controller reports Degraded=True: DegradedConditions: One or more other status conditions indicate a degraded state: DeploymentAvailable=False (DeploymentUnavailable: The deployment has Available status condition set to False (reason: MinimumReplicasUnavailable) with message: Deployment does not have minimum availability.), DeploymentReplicasMinAvailable=False (DeploymentMinimumReplicasNotMet: 0/2 of replicas are available, max unavailable is 1: Some pods are not scheduled: Pod "router-default-58d56594bb-g9bvt" cannot be scheduled: 0/3 nodes are available: 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/3 nodes are available: 3 Preemption is not helpful for scheduling.. Pod "router-default-58d56594bb-wsn8m" cannot be scheduled: 0/3 nodes are available: 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/3 nodes are available: 3 Preemption is not helpful for scheduling.. Make sure you have sufficient worker nodes.), CanaryChecksSucceeding=Unknown (CanaryRouteNotAdmitted: Canary route is not admitted by the default ingress controller) 
INFO Cluster operator ingress EvaluationConditionsDetected is False with AsExpected:  
INFO Cluster operator insights ClusterTransferAvailable is False with Disconnected: failed to pull cluster transfer: unable to retrieve cluster transfer data fr]$ ./openshift-install create cluster --dir ~/okdInstallation
INFO Consuming Install Config from target directory 
INFO Obtaining RHCOS image file from 'https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/37.20221127.3.0/x86_64/fedora-coreos-37.20221127.3.0-vmware.x86_64.ova?sha256=' 
INFO The file was found in cache: /home/okd/.cache/openshift-installer/image_cache/fedora-coreos-37.20221127.3.0-vmware.x86_64.ova. Reusing... 
INFO Creating infrastructure resources...         
INFO Waiting up to 20m0s (until 9:48AM) for the Kubernetes API at https://api.amds-kub.jhuapl.edu:6443... 
INFO API v1.26.4-2868+a7ee68b55354d8-dirty up     
INFO Waiting up to 30m0s (until 10:03AM) for bootstrapping to complete... 
WARNING Failed to extract host addresses: could not extract IP with bootstrap MOID:  
INFO Skipping VM console logs gather: no gather methods registered for "vsphere" 
INFO Pulling debug logs froopenshift-install-linux-4.13.0-0.okd-2023-07-09-062029m the bootstrap machine 
WARNING Unable to stat /home/okd/okdInstallation/serial-log-bundle-20230710100339.tar.gz, skipping 
ERROR Cluster operator authentication Degraded is True with IngressStateEndpoints_MissingSubsets::OAuthClientsController_SyncError::OAuthServerDeployment_PreconditionNotFulfilled::OAuthServerRouteEndpointAccessibleController_SyncError::OAuthServerServiceEndpointAccessibleController_SyncError::OAuthServerServiceEndpointsEndpointAccessibleController_SyncError::WellKnownReadyController_SyncError: IngressStateEndpointsDegraded: No subsets found for the endpoints of oauth-server 
ERROR OAuthClientsControllerDegraded: no ingress for host oauth-openshift.apps.amds-kub.jhuapl.edu in route oauth-openshift in namespace openshift-authentication 
ERROR OAuthServerDeploymentDegraded: waiting for the oauth-openshift route to contain an admitted ingress: no admitted ingress for route oauth-openshift in namespace openshift-authentication 
ERROR OAuthServerDeploymentDegraded:               
ERROR OAuthServerRouteEndpointAccessibleControllerDegraded: route "openshift-authentication/oauth-openshift": status does not have a valid host address 
ERROR OAuthServerServiceEndpointAccessibleControllerDegraded: Get "https://172.30.26.201:443/healthz": dial tcp 172.30.26.201:443: connect: connection refused 
ERROR OAuthServerServiceEndpointsEndpointAccessibleControllerDegraded: oauth service endpoints are not ready 
ERROR WellKnownReadyControllerDegraded: failed to get oauth metadata from openshift-config-managed/oauth-openshift ConfigMap: configmap "oauth-openshift" not found (check authentication operator, it is supposed to create this) 
ERROR Cluster operator authentication Available is False with OAuthServerDeployment_PreconditionNotFulfilled::OAuthServerServiceEndpointAccessibleController_EndpointUnavailable::OAuthServerServiceEndpointsEndpointAccessibleController_ResourceNotFound::ReadyIngressNodes_NoReadyIngressNodes::WellKnown_NotReady: OAuthServerServiceEndpointAccessibleControllerAvailable: Get "https://172.30.26.201:443/healthz": dial tcp 172.30.26.201:443: connect: connection refused 
ERROR OAuthServerServiceEndpointsEndpointAccessibleControllerAvailable: endpoints "oauth-openshift" not found 
ERROR ReadyIngressNodesAvailable: Authentication requires functional ingress which requires at least one schedulable and ready node. Got 0 worker nodes, 3 master nodes, 0 custom target nodes (none are schedulable or ready for ingress pods). 
ERROR WellKnownAvailable: The well-known endpoint is not yet available: failed to get oauth metadata from openshift-config-managed/oauth-openshift ConfigMap: configmap "oauth-openshift" not found (check authentication operator, it is supposed to create this) 
INFO Cluster operator baremetal Disabled is False with :  
INFO Cluster operator cloud-controller-manager TrustedCABundleControllerControllerAvailable is True with AsExpected: Trusted CA Bundle Controller works as expected 
INFO Cluster operator cloud-controller-manager TrustedCABundleControllerControllerDegraded is False with AsExpected: Trusted CA Bundle Controller works as expected 
INFO Cluster operator cloud-controller-manager CloudConfigControllerAvailable is True with AsExpected: Cloud Config Controller works as expected 
INFO Cluster operator cloud-controller-manager CloudConfigControllerDegraded is False with AsExpected: Cloud Config Controller works as expected 
INFO Cluster operator cloud-controller-manager CloudControllerOwner is True with AsExpected: Cluster Cloud Controller Manager Operator owns cloud controllers at 4.13.0-0.okd-2023-07-09-062029 
ERROR Cluster operator cluster-autoscaler Degraded is True with MissingDependency: machine-api not ready 
ERROR Cluster operator console Available is False with RouteHealth_RouteNotAdmitted: RouteHealthAvailable: console route is not admitted 
INFO Cluster operator etcd RecentBackup is Unknown with ControllerStarted: The etcd backup controller is starting, and will decide if recent backups are available or if a backup is required 
ERROR Cluster operator ingress Available is False with IngressUnavailable: The "default" ingress controller reports Available=False: IngressControllerUnavailable: One or more status conditions indicate unavailable: DeploymentAvailable=False (DeploymentUnavailable: The deployment has Available status condition set to False (reason: MinimumReplicasUnavailable) with message: Deployment does not have minimum availability.) 
INFO Cluster operator ingress Progressing is True with Reconciling: ingresscontroller "default" is progressing: IngressControllerProgressing: One or more status conditions indicate progressing: DeploymentRollingOut=True (DeploymentRollingOut: Waiting for router deployment rollout to finish: 0 of 2 updated replica(s) are available... 
INFO ).                                           
INFO Not all ingress controllers are available.   
ERROR Cluster operator ingress Degraded is True with IngressDegraded: The "default" ingress controller reports Degraded=True: DegradedConditions: One or more other status conditions indicate a degraded state: DeploymentAvailable=False (DeploymentUnavailable: The deployment has Available status condition set to False (reason: MinimumReplicasUnavailable) with message: Deployment does not have minimum availability.), DeploymentReplicasMinAvailable=False (DeploymentMinimumReplicasNotMet: 0/2 of replicas are available, max unavailable is 1: Some pods are not scheduled: Pod "router-default-58d56594bb-g9bvt" cannot be scheduled: 0/3 nodes are available: 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/3 nodes are available: 3 Preemption is not helpful for scheduling.. Pod "router-default-58d56594bb-wsn8m" cannot be scheduled: 0/3 nodes are available: 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/3 nodes are available: 3 Preemption is not helpful for scheduling.. Make sure you have sufficient worker nodes.), CanaryChecksSucceeding=Unknown (CanaryRouteNotAdmitted: Canary route is not admitted by the default ingress controller) 
INFO Cluster operator ingress EvaluationConditionsDetected is False with AsExpected:  
INFO Cluster operator insights ClusterTransferAvailable is False with Disconnected: failed to pull cluster transfer: unable to retrieve cluster transfer data from https://api.openshift.com/api/accounts_mgmt/v1/cluster_transfers/: Get "https://api.openshift.com/api/accounts_mgmt/v1/cluster_transfers/?search=cluster_uuid+is+%27ba30e135-e9f8-4a0f-b436-f082fa302f0f%27+and+status+is+%27accepted%27": x509: certificate signed by unknown authority 
INFO Cluster operator insights Disabled is False with AsExpected:  
INFO Cluster operator insights SCAAvailable is Unknown with :  
ERROR Cluster operator kube-apiserver Degraded is True with GuardController_SyncError: GuardControllerDegraded: Missing operand on node amds-kub-vxngj-master-0 
INFO Cluster operator kube-apiserver Progressing is True with NodeInstaller: NodeInstallerProgressing: 1 nodes are at revision 0; 2 nodes are at revision 7 
INFO Cluster operator kube-controller-manager Progressing is True with NodeInstaller: NodeInstallerProgressing: 3 nodes are at revision 7; 0 nodes have achieved new revision 8 
INFO Cluster operator machine-api Progressing is True with SyncingResources: Progressing towards operator: 4.13.0-0.okd-2023-07-09-062029 
ERROR Cluster operator machine-api Degraded is True with SyncingFailed: Failed when progressing towards operator: 4.13.0-0.okd-2023-07-09-062029 because minimum worker replica count (2) not yet met: current running replicas 0, waiting for [amds-kub-vxngj-worker-0-84rqf amds-kub-vxngj-worker-0-f6qkt amds-kub-vxngj-worker-0-fzrgc] 
ERROR Cluster operator machine-api Available is False with Initializing: Operator is initializing 
ERROR Cluster operator monitoring Available is False with UpdatingPrometheusOperatorFailed: reconciling Prometheus Operator Admission Webhook Deployment failed: updating Deployment object failed: waiting for DeploymentRollout of openshift-monitoring/prometheus-operator-admission-webhook: got 2 unavailable replicas 
ERROR Cluster operator monitoring Degraded is True with UpdatingPrometheusOperatorFailed: reconciling Prometheus Operator Admission Webhook Deployment failed: updating Deployment object failed: waiting for DeploymentRollout of openshift-monitoring/prometheus-operator-admission-webhook: got 2 unavailable replicas 
INFO Cluster operator monitoring Progressing is True with RollOutInProgress: Rolling out the stack. 
INFO Cluster operator network ManagementStateDegraded is False with :  
INFO Cluster operator network Progressing is True with Deploying: Deployment "/openshift-network-diagnostics/network-check-source" is waiting for other operators to become ready 
ERROR Bootstrap failed to complete: timed out waiting for the condition 
ERROR Failed to wait for bootstrapping to complete. This error usually happens when there is a problem with control plane hosts that prevents the control plane operators from creating the control plane. 
WARNING The bootstrap machine is unable to resolve API and/or API-Int Server URLs 
INFO Checking if api-int.amds-kub.jhuapl.edu of type API_INT_URL is resolvable 
INFO Starting stage resolve-api-int-url           
INFO Unable to resolve API_INT_URL api-int.amds-kub.jhuapl.edu 
INFO Bootstrap gather logs captured here "/home/okd/okdInstallation/log-bundle-20230710100339.tar.gz" 
om https://api.openshift.com/api/accounts_mgmt/v1/cluster_transfers/: Get "https://api.openshift.com/api/accounts_mgmt/v1/cluster_transfers/?search=cluster_uuid+is+%27ba30e135-e9f8-4a0f-b436-f082fa302f0f%27+and+status+is+%27accepted%27": x509: certificate signed by unknown authority 
INFO Cluster operator insights Disabled is False with AsExpected:  
INFO Cluster operator insights SCAAvailable is Unknown with :  
ERROR Cluster operator kube-apiserver Degraded is True with GuardController_SyncError: GuardControllerDegraded: Missing operand on node amds-kub-vxngj-master-0 
INFO Cluster operator kube-apiserver Progressing is True with NodeInstaller: NodeInstallerProgressing: 1 nodes are at revision 0; 2 nodes are at revision 7 
INFO Cluster operator kube-controller-manager Progressing is True with NodeInstaller: NodeInstallerProgressing: 3 nodes are at revision 7; 0 nodes have achieved new revision 8 
INFO Cluster operator machine-api Progressing is True with SyncingResources: Progressing towards operator: 4.13.0-0.okd-2023-07-09-062029 
ERROR Cluster operator machine-api Degraded is True with SyncingFailed: Failed when progressing towards operator: 4.13.0-0.okd-2023-07-09-062029 because minimum worker replica count (2) not yet met: current running replicas 0, waiting for [amds-kub-vxngj-worker-0-84rqf amds-kub-vxngj-worker-0-f6qkt amds-kub-vxngj-worker-0-fzrgc] 
ERROR Cluster operator machine-api Available is False with Initializing: Operator is initializing 
ERROR Cluster operator monitoring Available is False with UpdatingPrometheusOperatorFailed: reconciling Prometheus Operator Admission Webhook Deployment failed: updating Deployment object failed: waiting for DeploymentRollout of openshift-monitoring/prometheus-operator-admission-webhook: got 2 unavailable replicas 
ERROR Cluster operator monitoring Degraded is True with UpdatingPrometheusOperatorFailed: reconciling Prometheus Operator Admission Webhook Deployment failed: updating Deployment object failed: waiting for DeploymentRollout of openshift-monitoring/prometheus-operator-admission-webhook: got 2 unavailable replicas 
INFO Cluster operator monitoring Progressing is True with RollOutInProgress: Rolling out the stack. 
INFO Cluster operator network ManagementStateDegraded is False with :  
INFO Cluster operator network Progressing is True with Deploying: Deployment "/openshift-network-diagnostics/network-check-source" is waiting for other operators to become ready 
ERROR Bootstrap failed to complete: timed out waiting for the condition 
ERROR Failed to wait for bootstrapping to complete. This error usually happens when there is a problem with control plane hosts that prevents the control plane operators from creating the control plane. 
WARNING The bootstrap machine is unable to resolve API and/or API-Int Server URLs 
INFO Checking if api-int.amds-kub.jhuapl.edu of type API_INT_URL is resolvable 
INFO Starting stage resolve-api-int-url           
INFO Unable to resolve API_INT_URL api-int.amds-kub.jhuapl.edu 
INFO Bootstrap gather logs captured here "/home/okd/okdInstallation/log-bundle-20230710100339.tar.gz"

**Comment by vrutkovs** [comment]
> INFO Bootstrap gather logs captured here "/home/okd/okdInstallation/log-bundle-20230710100339.tar.gz"

Please attach (or upload to the public file sharing service) [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting)

**Comment by jgforbes** [comment]
[log-bundle-20230710100339.tar.gz](https://github.com/okd-project/okd/files/12003561/log-bundle-20230710100339.tar.gz)

**Reply by vrutkovs** [reply]
>Unable to connect to the server: dial tcp: lookup api-int.amds-kub.jhuapl.edu: no such host

Make sure you have created DNS records for `api.` and `api-int.`

**Reply by jgforbes** [reply]
I have for api., but the documentation does not reference api-int.
Is this a new requirement? The RH OpenShift installer works without it.

**Reply by vrutkovs** [reply]
Ah, I didn't notice you're using vsphere IPI, sorry.

The problem is
```
Jul 10 13:31:58 localhost.localdomain nm-dispatcher[1061]: <13>Jul 10 13:31:58 root: NM local-dns-prepender triggered by lo up.
Jul 10 13:31:58 localhost.localdomain nm-dispatcher[1060]: NM resolv-prepender: Setting up systemd-resolved for local DNS
Jul 10 13:31:58 localhost.localdomain nm-dispatcher[1067]: No value found
```

DNS server record in /etc/resolve.conf so local CoreDNS address could not be appended. How does this machine get network settings - DHCP or via vsphere metadata?

**Reply by jgforbes** [reply]
Via DHCP

**Reply by dastano98** [reply]
Hi how do you resolve it, i face the same issue for api

WARNING Unable to stat /root/ocp413/serial-log-bundle-20230910003656.tar.gz, skipping
ERROR Attempted to gather ClusterOperator status after installation failure: listing ClusterOperator objects: Get "https://api.ocp413.hc.xxxx.cloud:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 10.0.10.47:6443: connect: no route to host
ERROR Bootstrap failed to complete: Get "https://api.ocp413.hc.xxxxx.cloud:6443/version": dial tcp 10.0.10.47:6443: connect: no route to host
ERROR Failed waiting for Kubernetes API. This error usually happens when there is a problem on the bootstrap host that prevents creating a temporary control plane.

Please advised?
